dipy/denoise/denspeed.pyx

```diff
@@ -117,8 +111,8 @@ def _nlmeans_3d(double[:, :, ::1] arr, double[:, :, ::1] mask,
 
                     out[i, j, k] = process_block(arr, i, j, k, B, P, sigma)
 
-    if have_openmp and num_threads is not None:
-        openmp.omp_set_num_threads(all_cores)
+    if num_threads is not None:
+        restore_default_num_threads()
```

Comment:
Just out of curiosity, is it possible that this logic has unexpected side effects? What I understand is that we are "restoring" the number of threads after finishing the parallel block. But what if the number of threads was not the default when we called this function?